decode.go

```diff
@@ -18,30 +18,58 @@ type Unmarshaler interface {
 
 // Unmarshal parses the plist-encoded data and stores the result in the value pointed to by v.
 func Unmarshal(data []byte, v interface{}) error {
-	return NewDecoder(bytes.NewReader(data)).Decode(v)
+	// Check for binary plist here before setting up the decoder.
+	if string(data[:7]) == "bplist0" {
+		return NewBinaryDecoder(bytes.NewReader(data)).Decode(v)
+	}
+	return NewXMLDecoder(bytes.NewReader(data)).Decode(v)
 }
 
 // A Decoder reads and decodes Apple plist objects from an input stream.
+// The plists can be in XML or binary format.
 type Decoder struct {
-	reader io.Reader
+	reader   interface{} // io.Reader for XML, io.ReadSeeker for binary
+	isBinary bool        // true if this is a binary plist
+}
+
+// NewXMLDecoder returns a new decoder that reads an XML plist from r.
+func NewXMLDecoder(r io.Reader) *Decoder {
+	return &Decoder{reader: r, isBinary: false}
 }
 
-// NewDecoder returns a new decoder that reads from r.
-func NewDecoder(r io.Reader) *Decoder {
```

Comment:
NewDecoder should stay for backwards compatibility and default to XML. 
You can add a comment that the method is Deprecated.

Reply:
Another option is to keep a single `NewDecoder` method, but add a option like 
```
func NewDecoder(r io.Reader, opts ...Option) *Decoder
type Option func(*Decoder)
func Binary() Option {
  return func(d *Decoder) {
   d.isBinary = true
  } 
}
```

This change would allow the interface to stay backwards compatible and the config can asert what kind of reader is used.

Reply:
Added back NewDecoder which defaults to XML.  I'm wary of using NewDecoder for both xml and binary since the io.Reader argument would be misleading for binary decoders.